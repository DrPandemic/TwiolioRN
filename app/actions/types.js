// @flow

import { PhoneNumber, Message } from '../types';

export const SET_FETCHED_ACCOUNT_NUMBERS = 'SET_FETCHED_ACCOUNT_NUMBERS';
export const FETCH_ACCOUNT_NUMBERS = 'FETCH_ACCOUNT_NUMBERS';
export const FETCH_ACCOUNT_NUMBER_ERROR = 'FETCH_ACCOUNT_NUMBER_ERROR';

export const SELECT_NUMBER = 'SELECT_NUMBER';
export const SELECT_CONVERSATION = 'SELECT_CONVERSATION';

export const FETCH_MESSAGES = 'FETCH_MESSAGES';
export const SET_FETCHED_MESSAGES = 'SET_FETCHED_MESSAGES';
export const FETCH_MESSAGE_ERROR = 'FETCH_MESSAGE_ERROR';

export type FetchAccountNumbersT = {|
  type: 'FETCH_ACCOUNT_NUMBERS',
|}
export type SuccessFetchAccountNumbersT = {|
  type: 'SET_FETCHED_ACCOUNT_NUMBERS',
  fetchedAccountNumbers: Array<PhoneNumber>,
|}
export type FailFetchAccountNumbersT = {|
    type: 'FETCH_ACCOUNT_NUMBER_ERROR',
    error: any,
|}

export type SelectNumberT = {|
  type: 'SELECT_NUMBER',
  selectedNumber: ?string,
|}

export type FetchMessagesT = {|
  type: 'FETCH_MESSAGES',
|}
export type SuccessFetchMessagesT = {|
    type: 'SET_FETCHED_MESSAGES',
    fetchedMessages: Array<Message>,
|}
export type FailFetchMessagesT = {|
    type: 'FETCH_MESSAGE_ERROR',
    error: any,
|}

export type ActionT =
  | FetchAccountNumbersT
  | SuccessFetchAccountNumbersT
  | FailFetchAccountNumbersT
  | SelectNumberT
  | SelectConversationT
  | FetchMessagesT
  | SuccessFetchMessagesT
  | FailFetchMessagesT;
