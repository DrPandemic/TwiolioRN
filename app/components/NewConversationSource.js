// @flow

import React, { Component } from 'react';
import {
  Dimensions,
  FlatList,
  View,
  StyleSheet
} from 'react-native';
import { List, ListItem } from 'react-native-elements';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { ActionCreators } from '../actions';
import { expandPhoneNumbers } from '../types/Contact';
import { Colors } from '../constants';
import type { StateT } from '../reducers';
import type { T as ContactT } from '../reducers/contacts';
import type { PhoneNumber } from '../types/PhoneNumber';

const { width } = Dimensions.get('window');

const styles = StyleSheet.create({
  container: {
    width,
    flexDirection: 'column',
    backgroundColor: Colors.background,
  },
  list: {
    marginTop: 0,
    borderBottomWidth: 0,
  },
  item: {
    paddingTop: 10,
    paddingBottom: 10,
    borderBottomWidth: 0,
  },
  flatList: {
    marginBottom: 30,
  },
});

type PropsT = {
  numbers: Array<PhoneNumber>,
  push: (any) => void,
};

export class PNewConversationSource extends Component {
  props: PropsT;

  constructor(props: PropsT): void {
    super(props);

    this.onSelectContact = this.onSelectContact.bind(this);
  }

  onSelectContact(row: PhoneNumber): void {
    alert(JSON.stringify(row));
  }

  renderRow(row: PhoneNumber) {
    return (
      <ListItem
        title={row.friendlyName}
        onPress={() => this.onSelectContact(row)}
        containerStyle={styles.item}
        underlayColor={'#dedede'}
        hideChevron={true}
      />
    );
  }

  render() {
    return (
      <View style={styles.container}>
        <List style={styles.list}>
          <FlatList
            data={this.props.numbers}
            renderItem={({ item }) => this.renderRow(item)}
            keyExtractor={(row: PhoneNumber) => row.sid}
            style={styles.flatList}
          />
        </List>
      </View>
    );
  }
}

function mapStateToProps(state: StateT) {
  return {
    numbers: state.fetchedAccountNumbers.numbers,
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators(ActionCreators, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(PNewConversationSource);
