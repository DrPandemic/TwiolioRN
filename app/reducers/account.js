import createReducer from '../lib/createReducer'
import * as types from '../actions/types'

export const fetchAccountNumbers = createReducer({}, {
  [types.SET_ACCOUNT_NUMBER] (state, action) {
    return {
      ...state,
      accountNumbers: action.accountNumbers,
    };
  },
});

