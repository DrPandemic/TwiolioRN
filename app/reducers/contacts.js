// @flow

import { loop, Cmd } from 'redux-loop';
import RNContacts from 'react-native-contacts';


import { Contact } from '../types';
import effects from '../effects';
import * as types from '../actions/types';
import createReducer from '../lib/createReducer';
import * as contactActions from '../actions/contacts';
import { persistStore } from '../actions/persist';

export type T = {|
  loading: boolean,
  contacts: Array<Contact>,
  error: any,
|};
export const initialState: T = {
  loading: false,
  contacts: [],
  error: null,
};

export const reducer = createReducer({
  [types.TICK](state: T) {
    return loop(
      { ...state },
      Cmd.action(contactActions.fetchContacts())
    );
  },
  [types.FETCH_CONTACTS](
    state: T,
  ) {
    return loop(
      { ...state, loading: true },
      Cmd.run(effects.fetchContacts, {
        successActionCreator: contactActions.successFetchContacts,
        failActionCreator: contactActions.failFetchContacts,
        args: [RNContacts],
      })
    );
  },
  [types.SUCCESS_FETCH_CONTACTS](
    state: T,
    action: types.SuccessFetchContactsT,
  ) {
    return loop({
      ...state,
      contacts: action.contacts,
      loading: false,
      error: null,
    },
    Cmd.action(persistStore())
    );
  },
  [types.FAIL_FETCH_CONTACTS](
    state: T,
    action: types.FailFetchContactsT,
  ) {
    return loop({
      ...state,
      loading: false,
      error: action.error,
    },
    );
  },
});
