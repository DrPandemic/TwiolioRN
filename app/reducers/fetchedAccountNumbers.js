// @flow

import { loop, Cmd } from 'redux-loop';
import * as types from '../actions/types';
import * as fetchActions from '../actions/fetchedAccountNumbers';
import { getApi } from '../lib/api';
import createReducer from '../lib/createReducer';
import effects from '../effects';
import { PhoneNumber } from '../types';

export type T = {|
  loading: boolean,
  numbers: Array<PhoneNumber>,
  error: any,
|};
export const initialState: T = {
  loading: false,
  numbers: [],
  error: {},
};

export const reducer = createReducer({
  [types.TICK](state: T) {
    return loop(
      { ...state },
      Cmd.action(fetchActions.fetchAccountNumbers()),
    );
  },
  [types.FETCH_ACCOUNT_NUMBERS](state: T) {
    return loop(
      { ...state, loading: true },
      Cmd.run(effects.fetchNumbers, {
        successActionCreator: fetchActions.successFetchAccountNumbers,
        failActionCreator: fetchActions.failFetchAccountNumbers,
        args: [getApi()],
      })
    );
  },
  [types.SET_FETCHED_ACCOUNT_NUMBERS](
    state: T,
    action: types.SuccessFetchAccountNumbersT
  ) {
    return {
      ...state,
      numbers: action.fetchedAccountNumbers,
      error: {},
      loading: false,
    };
  },
  [types.FETCH_ACCOUNT_NUMBER_ERROR](
    state: T,
    action: types.FailFetchAccountNumbersT
  ) {
    return {
      ...state,
      error: action.error,
      loading: false,
    };
  },
});
