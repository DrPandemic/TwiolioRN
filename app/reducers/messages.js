// @flow

import { loop, Effects } from 'redux-loop';

import * as types from '../actions/types';
import type { ConversationStoreT } from '../types/ConversationStore';
import { addMessages } from '../types/ConversationStore';
import LibApi from '../lib/api';
import createReducer from '../lib/createReducer';
import effects from '../effects';

export type T = {|
  loading: boolean,
  messages: ConversationStoreT,
  error: any,
|};
export const initialState: T = {
  loading: false,
  messages: {},
  error: {},
};

export const reducer = createReducer({
  [types.FETCH_MESSAGES](state: T) {
    return loop(
      { ...state, loading: true },
      Effects.promise(effects.fetchMessages, LibApi)
    );
  },
  [types.SET_FETCHED_MESSAGES](
    state: T,
    action: types.SuccessFetchMessagesT
  ) {
    return {
      ...state,
      messages: addMessages(state.messages, action.fetchedMessages),
      error: {},
      loading: false,
    };
  },
  [types.FETCH_MESSAGE_ERROR](
    state: T,
    action: types.FailFetchMessagesT
  ) {
    return {
      ...state,
      error: action.error,
      loading: false,
    };
  },
});
