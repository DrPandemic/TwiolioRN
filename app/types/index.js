// @flow
import Message from './Message';
import PhoneNumber from './PhoneNumber';
import ConversationStore from './ConversationStore';

export {
  Message,
  PhoneNumber,
  ConversationStore,
};
