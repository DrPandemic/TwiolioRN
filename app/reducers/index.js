import { combineReducers } from 'redux';
import * as AccountReducers from './account'

export default combineReducers(Object.assign({},
  AccountReducers,
));
