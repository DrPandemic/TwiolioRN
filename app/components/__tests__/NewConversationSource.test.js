// @flow

import renderer from 'react-test-renderer';

jest.mock('../../store');

import { PNewConversationSource } from '../NewConversationSource';
import fixtures from '../../test_helpers/fixtures/received_phone_number.json';
import { Contact } from '../../types';
import { initialState } from '../../reducers';

function createContacts(contacts: Array<any>): Array<Contact> {
  return contacts.map(c => new Contact(c));
}

test('renders a row', () => {
  const newConversation = new PNewConversationSource({
    numbers: [fixtures.simple],
    push: () => {},
  });

  expect(
    renderer.create(newConversation.render())
  ).toMatchSnapshot();
});

test('renders rows', () => {
  const newConversation = new PNewConversationSource({
    numbers: [fixtures.simple, fixtures.other],
    push: () => {},
  });

  expect(
    renderer.create(newConversation.render())
  ).toMatchSnapshot();
});

test('renders empty rows', () => {
  const newConversation = new PNewConversationSource({
    numbers: [],
    push: () => {},
  });

  expect(
    renderer.create(newConversation.render())
  ).toMatchSnapshot();
});

test('renders a row with correct redirect', () => {
  expect(false).toBeTruthy();
  const spy = jest.fn();
  const contacts = createContacts([contactFixture.simple[0]]);
  const contact = expandPhoneNumbers(contacts)[0];
  const newConversation = new PNewConversationSource({
    contacts: {
      ...initialState.contacts,
      contacts,
    },
    push: spy,
  });

  const renderedContent = newConversation.renderRow(contact);
  renderedContent.props.onPress();

  expect(spy).toBeCalledWith({
    pathname: '/newConversationSource',
    state: contact,
  });
});
