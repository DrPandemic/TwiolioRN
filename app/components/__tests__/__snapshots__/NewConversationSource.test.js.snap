// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a row 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#eee",
      "flexDirection": "column",
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomWidth": 0,
        "marginTop": 0,
      }
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "account_sid": "account_sid",
            "address_requirements": "none",
            "api_version": "2010-04-01",
            "beta": false,
            "capabilities": Object {
              "fax": true,
              "mms": true,
              "sms": true,
              "voice": true,
            },
            "date_created": "Thu, 11 May 2017 04:05:46 +0000",
            "date_updated": "Fri, 12 May 2017 02:06:29 +0000",
            "emergency_address_sid": null,
            "emergency_status": "Inactive",
            "friendly_name": "(123) 123-1234",
            "phone_number": "+1231231234",
            "sid": "sid",
            "sms_application_sid": "",
            "sms_fallback_method": "POST",
            "sms_fallback_url": "",
            "sms_method": "POST",
            "sms_url": "https://handler.twilio.com/twiml/something",
            "status_callback": "",
            "status_callback_method": "POST",
            "trunk_sid": null,
            "uri": "/2010-04-01/Accounts/something/IncomingPhoneNumbers/somethingelse.json",
            "voice_application_sid": "",
            "voice_caller_id_lookup": false,
            "voice_fallback_method": "POST",
            "voice_fallback_url": "",
            "voice_method": "POST",
            "voice_receive_mode": "voice",
            "voice_url": "https://demo.twilio.com/welcome/voice/",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      onViewableItemsChanged={undefined}
      renderItem={[Function]}
      renderScrollComponent={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "marginBottom": 30,
        }
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderBottomColor": "#bbb",
                    "borderBottomWidth": 1,
                    "paddingBottom": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  },
                  Object {
                    "borderBottomWidth": 0,
                    "paddingBottom": 10,
                    "paddingTop": 10,
                  },
                ],
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                    "marginLeft": 10,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={undefined}
                >
                  <View />
                </View>
                <View
                  style={undefined}
                >
                  <View />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`renders empty rows 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#eee",
      "flexDirection": "column",
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomWidth": 0,
        "marginTop": 0,
      }
    }
  >
    <RCTScrollView
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      onViewableItemsChanged={undefined}
      renderItem={[Function]}
      renderScrollComponent={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "marginBottom": 30,
        }
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;

exports[`renders rows 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#eee",
      "flexDirection": "column",
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomWidth": 0,
        "marginTop": 0,
      }
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "account_sid": "account_sid",
            "address_requirements": "none",
            "api_version": "2010-04-01",
            "beta": false,
            "capabilities": Object {
              "fax": true,
              "mms": true,
              "sms": true,
              "voice": true,
            },
            "date_created": "Thu, 11 May 2017 04:05:46 +0000",
            "date_updated": "Fri, 12 May 2017 02:06:29 +0000",
            "emergency_address_sid": null,
            "emergency_status": "Inactive",
            "friendly_name": "(123) 123-1234",
            "phone_number": "+1231231234",
            "sid": "sid",
            "sms_application_sid": "",
            "sms_fallback_method": "POST",
            "sms_fallback_url": "",
            "sms_method": "POST",
            "sms_url": "https://handler.twilio.com/twiml/something",
            "status_callback": "",
            "status_callback_method": "POST",
            "trunk_sid": null,
            "uri": "/2010-04-01/Accounts/something/IncomingPhoneNumbers/somethingelse.json",
            "voice_application_sid": "",
            "voice_caller_id_lookup": false,
            "voice_fallback_method": "POST",
            "voice_fallback_url": "",
            "voice_method": "POST",
            "voice_receive_mode": "voice",
            "voice_url": "https://demo.twilio.com/welcome/voice/",
          },
          Object {
            "account_sid": "account_sid1",
            "address_requirements": "none",
            "api_version": "2010-04-01",
            "beta": false,
            "capabilities": Object {
              "fax": true,
              "mms": true,
              "sms": true,
              "voice": true,
            },
            "date_created": "Thu, 11 May 2017 04:05:46 +0000",
            "date_updated": "Fri, 12 May 2017 02:06:29 +0000",
            "emergency_address_sid": null,
            "emergency_status": "Inactive",
            "friendly_name": "(123) 123-1235",
            "phone_number": "+1231231235",
            "sid": "sid1",
            "sms_application_sid": "",
            "sms_fallback_method": "POST",
            "sms_fallback_url": "",
            "sms_method": "POST",
            "sms_url": "https://handler.twilio.com/twiml/something",
            "status_callback": "",
            "status_callback_method": "POST",
            "trunk_sid": null,
            "uri": "/2010-04-01/Accounts/something/IncomingPhoneNumbers/somethingelse.json",
            "voice_application_sid": "",
            "voice_caller_id_lookup": false,
            "voice_fallback_method": "POST",
            "voice_fallback_url": "",
            "voice_method": "POST",
            "voice_receive_mode": "voice",
            "voice_url": "https://demo.twilio.com/welcome/voice/",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      onViewableItemsChanged={undefined}
      renderItem={[Function]}
      renderScrollComponent={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "marginBottom": 30,
        }
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderBottomColor": "#bbb",
                    "borderBottomWidth": 1,
                    "paddingBottom": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  },
                  Object {
                    "borderBottomWidth": 0,
                    "paddingBottom": 10,
                    "paddingTop": 10,
                  },
                ],
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                    "marginLeft": 10,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={undefined}
                >
                  <View />
                </View>
                <View
                  style={undefined}
                >
                  <View />
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderBottomColor": "#bbb",
                    "borderBottomWidth": 1,
                    "paddingBottom": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  },
                  Object {
                    "borderBottomWidth": 0,
                    "paddingBottom": 10,
                    "paddingTop": 10,
                  },
                ],
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                    "marginLeft": 10,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={undefined}
                >
                  <View />
                </View>
                <View
                  style={undefined}
                >
                  <View />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
